src/css-select-parse5-adapter.ts

```diff
@@ -0,0 +1,94 @@
+import {Adapter as CSSSelectAdapter} from 'css-select';
+import {Attribute, Element, Node, TreeAdapter} from 'parse5';
+const defaultTreeAdapter = require('parse5/lib/tree-adapters/default');
+export type Predicate = (node: Node) => boolean;
+export class Adapter implements CSSSelectAdapter<Node, Element> {
+  treeAdapter: TreeAdapter;
+  constructor(treeAdapter: TreeAdapter) {
+    this.treeAdapter = treeAdapter;
+  }
+  isTag(node: Node): node is Element {
+    return this.treeAdapter.isElementNode(node);
+  }
+  existsOne(test: Predicate, elems: Element[]): boolean {
+    return elems.some(test);
+  }
+  getAncestors(node: Node): Node[] {
+    const parentNode = this.hasParent(node) && this.getParent(node);
```

Comment:
Is this necessary? Does `getParent` throw if there isn't one?

Reply:
Upon review this is not necessary as `this.getParentNode()` and `defaultTreeAdapter.getParentNode()` don't throw.  The `css-select` typings are wrong for this one, since `undefined` is a possible return value.  I was trying to work around that but there's no reason to test with hasParent since the ternary will handle undefined.  Removed hasParent (was my own method) and removed the test.